Mathlib/Logic/Function/Iterate.lean

```diff
@@ -54,6 +54,12 @@ theorem iterate_zero : f^[0] = id :=
   rfl
 #align function.iterate_zero Function.iterate_zero
 
+-- can be proved by simp but this is shorter and more natural
+@[simp, nolint simpNF]
```

Comment:
I think maybe the preference here should be for `simp high` (if that works) instead of `nolint simpNF`. Also, why did this move up 25 or so lines? If there is no reason, let's move it back to reduce the diff.